### PR TITLE
[TASK] Update composer dependencies

### PR DESCRIPTION
**Pull request by liayn**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,6 @@ language: php
 sudo: false
 
 php:
- - 5.5
```
Comment (compeak) [inline .travis.yml]: Please explain this.

Comment (liayn) [inline .travis.yml]: PHPunit 5.2 does not support PHP  5.5 anymore. That's ok for running the tests IMO.

Comment (compeak) [inline .travis.yml]: > PHPunit 5.2 does not support PHP 5.5 anymore.

[But we do](https://github.com/heiseonline/shariff-backend-php#requirements). I don't think it's a good idea to stop testing a supported version of PHP.

> That's ok for running the tests IMO.

PHP 5.5 ist also used to deploy the builds.

Comment (liayn) [inline .travis.yml]: Well, phpunit 4.8 does not support PHP7 actually, which we support as well.
That's why I opted for going with phpunit 5.2 and drop php5.5 testing support, since 5.6 and 5.5 are more or less the same anyway.

Comment (compeak) [inline .travis.yml]: So the most consistent thing to do would be
- marking PHP 5.5 as untested and "most likely working but not supported"
- updating phpunit and starting to do tests using PHP 7
- doing builds on PHP 5.6
?

Comment (liayn) [inline .travis.yml]: - We do not use any feature that is different in 5.5 and 5.6
- Our tests do not cover enough functions to be able at all to find any possible PHP bugs which might be in 5.5 but not in 5.6
- We need PHP for the builds only for composer. I would do that with the fastest one, that is PHP 7.
- We already test on PHP 7.

**Comment by liayn** [comment]
Rebased

**Comment by liayn** [comment]
Just realized that since today phpunit 5.3 is out and already included in the rebase. Will update the commit message.

File: tests/ShariffTest.php [hunk 2]
```diff
@@ -83,21 +82,6 @@ public function testInvalidDomain()
         $this->assertNull($counts);
     }
 
-    public function testApcCache()
```
Comment (core23) [inline tests/ShariffTest.php]: Why did you remove this test?

Comment (liayn) [inline tests/ShariffTest.php]: 1.) it failed on travis with PHP7 and phpunit 5.2+
2.) It does not make sense to test whether something is (not) loaded in the testing environment.

So instead of trying to fix the failing test, I simply removed it.

File: .travis.yml (hunk 1 above)
Comment (compeak) [inline .travis.yml]: Support for PHP 5.5 will be dropped, since EOL is only a few months away.

Comment (compeak) [inline .travis.yml]: PHP 5.5 is now EOL.

**Comment by compeak** [comment]
Thank you.
